### PR TITLE
Support for YouTube

### PR DESCRIPTION
**Pull request by antonmolodykh**
Refs https://github.com/Screenly/screenly-ose/issues/586

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -20,3 +20,8 @@ uptime==2.0.2
 werkzeug==0.10.4
 wsgiref==0.1.2
 futures==3.0.5
+youtube_dl==2017.7.2
```
Comment (vpetersson) [inline requirements.txt]: We should probably not pin this one. At the very least use `>=`.

File: server.py [hunk 2]
```diff
@@ -151,12 +153,20 @@ def get(key):
...
+        asset['is_processing'] = 1
+
     asset['uri'] = uri
 
     if "video" in asset['mimetype']:
-        video_duration = get_video_duration(asset['uri'])
-        if video_duration:
-            asset['duration'] = int(video_duration.total_seconds())
+        if asset['is_processing'] == 0:
+            video_duration = get_video_duration(uri)
```
Comment (vpetersson) [inline server.py]: Don't we get this from `youtube_dl`? It's a pretty heavy operation on the Pi, so we don't want to do it unless we need to.

Comment (antonmolodykh) [inline server.py]: We can get information about the duration of the video from youtube_dl, but it does not make much sense: after downloading a video from YouTube, we work with the asset as with a regular video asset (https://github.com/Screenly/screenly-ose/commit/c07d82319cff7b7bfbcafe607c871a378d98c456#diff-d355af669766ee3562643e8e5034f715R158). This means that when editing an asset, for example, changing the date - in any case, `get_video_duration` will be called.

Comment (vpetersson) [inline server.py]: But the duration is immutable. It should never change. The only video asset type it could change for would be for RTSP assets.

Comment (antonmolodykh) [inline server.py]: I understand it. But now this is how the processing of any video asset works: when adding or editing, the `get_video_duratio`n function works. I just made a video from YouTube a simple video asset.

Comment (vpetersson) [inline server.py]: Alright, please create a ticket about that then.

**Pull request merged**